Update extension-grants.rst with example

ionizer opened the pull request:
Updated documentation to describe how to implement custom grant types

ionizer committed:
Update extension-grants.rst with example

Updated documentation to describe how to implement custom grant types

ionizer committed:
Update extension-grants.rst wih example, fixed

Fixed a minor code error in the example after adding how-to

maxtruxa reviewed (changes requested):
Thanks for the PR. Could you look at the suggested changes?

ionizer commented on docs/misc/extension-grants.rst:
```diff
@@ -2,7 +2,52 @@
 
+.. code-block:: js
+
+  let AbstractGrantType = require('oauth2-server/lib/grant-types/abstract-grant-type');
```
Sorry for the late response. I wasn't aware about this when making my example. Thanks.

ionizer commented on docs/misc/extension-grants.rst:
```diff
@@ -2,7 +2,52 @@
+
+Extension grants are registered through `OAuth2Server#token() <https://oauth2-server.readthedocs.io/en/latest/api/oauth2-server.html#token-request-response-options-callback>`_ (``options.extendedGrantTypes``).
+
+This might require you to add the new ``grant_type`` into the ``client`` data
```
Sorry for being not so good at explaining this. But at least in my implementation, I perform checks against my ``client``'s ``grant_type`` data in the database, whether the grant type is valid or not. I'm still wondering how to best explain this.

ionizer commented on docs/misc/extension-grants.rst:
```diff
@@ -2,7 +2,52 @@
+
+  module.exports = MyCustomGrantType;
+
+Extension grants are registered through `OAuth2Server#token() <https://oauth2-server.readthedocs.io/en/latest/api/oauth2-server.html#token-request-response-options-callback>`_ (``options.extendedGrantTypes``).
```
I've changed it back the way it was

ionizer commented on docs/misc/extension-grants.rst:
```diff
@@ -2,7 +2,52 @@
+              refreshToken: await this.generateRefreshToken(client, user, scope),
+              refreshTokenExpiresAt: this.getRefreshTokenExpiresAt(),
+              scope: scope
+          }
```
Should have fixed it on my new commit.

ionizer committed:
Update documentation with requested changes

- No longer uses deep imports
- Fixed a missing semicolon
- Changed ref for OAuth2Server#token()
- Slightly elaboration on requirement to add a new ``grant_type``
